src/main.rs

```diff
@@ -6,122 +6,145 @@ mod node;
 use std::{
     collections::VecDeque,
     path::PathBuf,
     sync::{Arc, Mutex},
 };
-use structopt::{clap::ArgGroup, StructOpt};
+use structopt::StructOpt;
 use tokio::time::{interval_at, timeout, Duration, Instant};
 
-#[derive(StructOpt)]
-#[structopt(group = ArgGroup::with_name("testing").requires("test").multiple(true))]
+#[derive(StructOpt, Debug)]
 struct App {
     #[structopt(
         long = "node",
         help = "location of the GRPC interface of the node.",
         default_value = "http://localhost:10000",
-        env = "EURO2CCD_SERVICE_NODE"
+        env = "EUR2CCD_SERVICE_NODE"
     )]
     endpoint: endpoints::Endpoint,
     #[structopt(
         long = "rpc-token",
         help = "GRPC interface access token for accessing the node.",
         default_value = "rpcadmin",
-        env = "EURO2CCD_SERVICE_RPC_TOKEN"
+        env = "EUR2CCD_SERVICE_RPC_TOKEN"
     )]
     token: String,
     #[structopt(
         long = "secret-names",
         help = "Secret names on AWS to get govenance keys from.",
-        env = "EURO2CCD_SERVICE_SECRET_NAMES",
-        parse(from_str),
+        env = "EUR2CCD_SERVICE_SECRET_NAMES",
         use_delimiter = true,
-        default_value = "secret-dummy",
         required_unless = "local-keys",
         conflicts_with = "local-keys"
     )]
     secret_names: Vec<String>,
+    #[structopt(
+        long = "bitfinex-cert",
+        help = "Location of the bitfinex certificate file.",
+        env = "EUR2CCD_SERVICE_BITFINEX_CERTIFICATE",
+        default_value = config::BITFINEX_CERTIFICATE_LOCATION,
+        conflicts_with = "test",
+    )]
+    bitfinex_cert: PathBuf,
+    #[structopt(
+        long = "aws-region",
+        help = "Which AWS region to get the keys from.",
+        env = "EUR2CCD_SERVICE_AWS_REGION",
+        default_value = config::AWS_REGION,
+        conflicts_with = "local-keys",
+    )]
+    region: String,
     #[structopt(
         long = "update-interval",
         help = "How often to update the exchange rate. (In seconds)",
-        env = "EURO2CCD_SERVICE_UPDATE_INTERVAL",
+        env = "EUR2CCD_SERVICE_UPDATE_INTERVAL",
         default_value = "1800"
     )]
-    update_interval: u64,
+    update_interval: u32,
     #[structopt(
         long = "pull-exchange-interval",
         help = "How often to pull new exchange rate from exchange. (In seconds)",
-        env = "EURO2CCD_SERVICE_PULL_INTERVAL",
+        env = "EUR2CCD_SERVICE_PULL_INTERVAL",
         default_value = "60"
     )]
-    pull_interval: u64,
+    pull_interval: u32,
     #[structopt(
         long = "conversion_threshold_denominator",
         help = "Denominator for fraction that determines how far exchange rate can deviate from \
                 actual (bigint) value. (the numerator is 1)",
-        env = "EURO2CCD_SERVICE_CONVERSION_TRESHOLD_DENOMINATOR",
+        env = "EUR2CCD_SERVICE_CONVERSION_THRESHOLD_DENOMINATOR",
```

Comment:
I don't understand why, but github won't let me respond to [the previous comment](https://github.com/Concordium/concordium-euro2ccd-service/pull/2#discussion_r798811790).

The reason for adding the parameter is to terminate the approximation within a reasonable time frame. (It might be that the algorithm needs to be optimized)
The value I chose based on when the testcases ran at a "reasonable" time frame. (~a minute)

Reply:
Hmm, OK, I'll have a look at that in more detail. Perhaps I misunderstood the algorithm, but I thought it should terminate quickly.